pkg/provider/migration/legacy-api/machine/types.go

```diff
@@ -0,0 +1,1221 @@
+	// or deleted.
+	MachineDeploymentFrozen MachineDeploymentConditionType = "Frozen"
+)
+
+// MachineDeploymentCondition describes the state of a MachineDeployment at a certain point.
+type MachineDeploymentCondition struct {
+	// Type of MachineDeployment condition.
+	Type MachineDeploymentConditionType
+
+	// Status of the condition, one of True, False, Unknown.
+	Status ConditionStatus
+
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time
+
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+
+	// The reason for the condition's last transition.
+	Reason string
+
+	// A human readable message indicating details about the transition.
+	Message string
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineDeploymentList is a list of MachineDeployments.
+type MachineDeploymentList struct {
+	metav1.TypeMeta
+	// Standard list metadata.
+	metav1.ListMeta
+
+	// Items is the list of MachineDeployments.
+	Items []MachineDeployment
+}
+
+// describes the attributes of a scale subresource
+type ScaleSpec struct {
+	// desired number of machines for the scaled object.
+	Replicas int32
+}
+
+// represents the current status of a scale subresource.
+type ScaleStatus struct {
+	// actual number of observed machines of the scaled object.
+	Replicas int32
+
+	// label query over machines that should match the replicas count. More info: http://kubernetes.io/docs/user-guide/labels#label-selectors
+	Selector *metav1.LabelSelector
+
+	// label selector for machines that should match the replicas count. This is a serializated
+	// version of both map-based and more expressive set-based selectors. This is done to
+	// avoid introspection in the clients. The string will be in the same format as the
+	// query-param syntax. If the target type only supports map-based selectors, both this
+	// field and map-based selector field are populated.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	TargetSelector string
+}
+
+// +genclient
+// +genclient:noVerbs
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// represents a scaling request for a resource.
+type Scale struct {
+	metav1.TypeMeta
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata.
+	metav1.ObjectMeta
+
+	// defines the behavior of the scale. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status.
+	Spec ScaleSpec
+
+	// current status of the scale. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status. Read-only.
+	Status ScaleStatus
+}
+
+/********************** OpenStackMachineClass APIs ***************/
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OpenStackMachineClass TODO
+type OpenStackMachineClass struct {
```

Comment:
Sorry to comment late, but why do we need this ?

Reply:
It's for the migration path. It's just a copy of the this file of MCM fork. We will actually only need the `MetalMachineClass` so it would be possible to cleanup the other providers. The old definition is required in order to map from the prior resource to `MachineClass`.